pkg/splunk/enterprise/clustermaster.go

```diff
@@ -85,6 +85,7 @@ func ApplyClusterMaster(client splcommon.ControllerClient, cr *enterprisev1.Clus
 
 	// check if deletion has been requested
 	if cr.ObjectMeta.DeletionTimestamp != nil {
+		DeleteOwnerReferencesForS3SecretObjects(client, cr, &cr.Spec.SmartStore)
```

Comment:
Have a generic API, so that other calls can be added as part of that API, and then from there call deleting the references for S3 ssecrets